### PR TITLE
Upgrade to upstream 1.25.0 and sync Helm chart for missing features

### PR DESCRIPTION
**Pull request by primeroz**
Towards: https://github.com/giantswarm/giantswarm/issues/24865

- Bumped `azuredisk-csi` to upstream version 1.25.0
- Increased qps limits for csi-provisioner and csi-attacher to match upstream
- Increased WorkerThreads for csi-provisioner to match upstream
- Increased csi-attacher timeout from 600s to 1200s to match upstream
- Add support for http/s proxy settings to azuredisk
- as per [KEP 2067](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cluster-lifecycle/kubeadm/2067-rename-master-label-taint/README.md#renaming-the-node-rolekubernetesiomaster-node-label) add toleration for `node-role.kubernetes.io/control-plane` label and update nodeSelector
- add rbac to get nodes for azuredisk-node to match upstream
- Make some hardcoded fields in the template a value so that we can set them differently in `Vintage` and in `CAPZ`
  - runOnMaster nodeSelector label
  - AZURE_CREDENTIAL_FILE location

---
Summary of Changes from upstream project
#### 1.22

* chart: As per [KEP 2067](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cluster-lifecycle/kubeadm/2067-rename-master-label-taint/README.md), `node-role.kubernetes.io/master` taint is replaced with `node-role.kubernetes.io/control-plane`
* Bugfix for default iops on UltraSSD disks

#### 1.23

* support PremiumV2_LRS disk type
* support location in storage class
* chart: support http/s proxy settings
* increase csi-attaccher timeout

#### 1.24

* chart: increase client QPS settings for csi-attacher , csi-provisioner

#### 1.25
* support advanced perf profile on linux nodes
* chart: add no_proxy settings

--- 
ToDo:
  - [x] Check `node-role.kubernetes.io/control-plane` on a Workload Cluster azure vintage that the label is set on master node 
    - Vintage MC cluster do have the label but WC do not. i will make this a value in the values.yaml and default to old value
  - [x] Check `node-role.kubernetes.io/control-plane` on _glippy_ that label is set on master node
  - [x]  Check location and contents of `AZURE_CREDENTIAL_FILE` in vintage vs Capz
    - vintage: /etc/kubernetes/config/azure.yaml
    - capz: /etc/kubernetes/azure.json
    - [x] Support having different values for `AZURE_CREDENTIAL_FILE` in _vintage_ vs _capz_

---
Tests:
- [x] Vintage WC - on Godsmack
  - [x] Check Rendered Yaml `helm get manifest` and compare to current version
  - [x] Create Volume
  - [x] Delete volume
  - [x] Create Snapshot
  - [x] Restore Snapshot
- [x] CAPZ 
  - [x] Create Volume
  - [x] Delete Volume
  - [x] Create Snapshot
  - [x] Restore Snapshot

### CODE REVIEW

**Comment by primeroz** [comment]
I expected that something is not quite right with this PR since it was pushed to the test catalog as `"add azuredisk-csi-driver-app-1.21.0-gs4-67d2032765d14fb6dd904c777c4221c1e7487ccb.tgz"` rather than 1.25.0 ... need to figure out how to get the right version

**Review by calvix — APPROVED** [review]
lgtm if @whites11 is happy

File: helm/azuredisk-csi-driver-app/values.yaml
Comment (whites11) [inline helm/azuredisk-csi-driver-app/values.yaml:56]: I'm not 100% sure this will work on capi. But nevermind we can always fix it afterwards

**Review by whites11 — APPROVED** [review]
let's go

File: helm/azuredisk-csi-driver-app/values.yaml
Comment (primeroz) [reply helm/azuredisk-csi-driver-app/values.yaml:56]: yeah in CAPI i need to change that and the azurefile credential location, that's why i moved both to the values but left the default to match what is currently used in vintage 

https://github.com/giantswarm/default-apps-azure/pull/7/commits/fd73d68203f33ccf614c536a14a222d76f010a2b#diff-9cfdd527b22ba84cdc4ad1df7c0e647c13dea797bc547e867476568f374778fd

Comment (whites11) [reply helm/azuredisk-csi-driver-app/values.yaml:56]: Gr8 thx for explaining

Comment (calvix) [reply helm/azuredisk-csi-driver-app/values.yaml:56]: that's why its value that can be overridden in `default-apps-azure` :)

**Comment by primeroz** [comment]
FYI i am having an issue on capz i need to resolve, and probalby retest on vintage as well 
https://kubernetes.slack.com/archives/CEX9HENG7/p1669828285278899

so i will probably have to apply some changes and re-ask for reviews eventually , sorry i rushed it :) :pray:

**Comment by primeroz** [comment]
Issue was resolved by setting `disable-avset-nodes=false` , which it has been set to false upstream by default since 1.18.0 https://github.com/kubernetes-sigs/azuredisk-csi-driver/commit/9a1b96c1b4973bc329a9dd4f30816ba9925ee789 

I will try to understand what does it acually do then we should decide what to do with default setting for legacy ( which is set to true )  @whites11

**Review by whites11 — DISMISSED** [review]

**Comment by primeroz** [comment]
Upstream description of the flag 

```
that's an optimization if it's set as true, it would skip avset vm list check directly, it's ok to set as false by default, which may cause a little more single vm list operation on VMSS clusters, but that's ok
```

I will do a second test on vintage , i already did one with the flag as it is now and it worked fine at least on a brand new vintage cluster i created on purpose wich runs both master and workers as `VMSS`

**Comment by primeroz** [comment]
Asking for re-review due to some code changes and because this is my first time pushing code to an app :) 

@giantswarm/team-phoenix i did get a review from @whites11 before but he is off now so i just added the whole team back as a reviewer since this does impact you to some extent. 

I did all i think was needed to keep compatibility between vintage and capz for the helm chart. 

I also tested the upgraded chart on a Vintage WC but not a MC 

From what i understand even when released this won't apply to any vintage cluster until is added to a new release for WC or updated in Terraform for MC ? 

Thanks

**Review by calvix — APPROVED** [review]

File: helm/azuredisk-csi-driver-app/templates/azuredisk-controller/deployment.yaml
Comment (primeroz) [inline helm/azuredisk-csi-driver-app/templates/azuredisk-controller/deployment.yaml:30]: this is not working as expected , on CAPZ cluster where we do set these values

https://github.com/giantswarm/default-apps-azure/blob/add_values_json_schema/helm/default-apps-azure/values.yaml#L7

I end up with ( from helm get all azuredisk-csi-driver )

```
  masterNodeSelector:
    kubernetes.io/role: master
    node-role.kubernetes.io/control-plane: ""
```
and

```
      nodeSelector:
        kubernetes.io/os: linux
        kubernetes.io/role: master
        node-role.kubernetes.io/control-plane: 
```

in the actual deployment yaml

which, i guess due to the lack of `""` as a value ends up as 

```
     nodeSelector:
        kubernetes.io/os: linux
        kubernetes.io/role: master
```

in the actual deployment in cluster
which is not what i expected , i only expected the `node-role` one

Comment (primeroz) [reply helm/azuredisk-csi-driver-app/templates/azuredisk-controller/deployment.yaml:30]: https://github.com/giantswarm/default-apps-azure/pull/7/commits/b4acfb1bdb460df8fcb8ba408cbf397685bf5023

_fixes_ the issue but is quite ugly

Comment (primeroz) [reply helm/azuredisk-csi-driver-app/templates/azuredisk-controller/deployment.yaml:30]: actually this seems to be the way to this by `nulling` the value in the chart values.yaml 

https://helm.sh/docs/chart_template_guide/values_files/#deleting-a-default-key

unless we restructure the template to something like

```
nodeSelector:
{{- if vintage }}
   kubernetes.io/role: master
{{- else if capz }}
  node-role.kubernetes.io/control-plane: ""
{{ - else }}
  ERROR unsupported 
{{- end }}
```

which would require having `knowledge` of the installation type in the values and would require modifying the template for every possible value of nodeSelector rather than having it configured as a value.

I will keep the fix as it is ( so it does not need any change on vintage values ) but will keep this conversation open for now if anyone want to comment on it

**Review by T-Kukawka — APPROVED** [review]

**Review by bavarianbidi — APPROVED** [review]

**Pull request merged**
